app/src/main/java/org/mozilla/fenix/home/recentbookmarks/view/RecentBookmarksHeaderViewHolder.kt

```diff
@@ -18,26 +29,52 @@ import org.mozilla.fenix.utils.view.ViewHolder
  * @param interactor [RecentBookmarksInteractor] which will have delegated to all user interactions.
  */
 class RecentBookmarksHeaderViewHolder(
-    view: View,
+    composeView: ComposeView,
+    viewLifecycleOwner: LifecycleOwner,
     private val interactor: RecentBookmarksInteractor
-) : ViewHolder(view) {
+) : ComposeViewHolder(composeView, viewLifecycleOwner) {
 
     init {
-        val binding = RecentBookmarksHeaderBinding.bind(view)
-        binding.showAllBookmarksButton.setOnClickListener {
-            dismissSearchDialogIfDisplayed()
-            interactor.onShowAllBookmarksClicked()
-        }
+        val horizontalPadding =
+            composeView.resources.getDimensionPixelSize(R.dimen.home_item_horizontal_margin)
+        composeView.setPadding(horizontalPadding, 0, horizontalPadding, 0)
     }
 
-    private fun dismissSearchDialogIfDisplayed() {
+    private fun dismissSearchDialogIfDisplayedAndShowAllClicked() {
```

Comment:
I don't think we should combine the 2. Let's just keep the call similar to before.
```
onShowAllClick = {
            dismissSearchDialogIfDisplayed()
            interactor.onShowAllBookmarksClicked()
}
```